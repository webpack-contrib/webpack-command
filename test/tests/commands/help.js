const { test } = require('../../util');
const HelpCommand = require('../../../lib/commands/HelpCommand');
const {
  help: commandHelp,
  load: getCommands,
} = require('../../../lib/commands/');

let command;
const commands = getCommands();
const { HelpCommandError } = HelpCommand;

test('help command', module, () => {
  it(`should instantiate`, () => {
    command = new HelpCommand();
    expect(command).toBeInstanceOf(HelpCommand);
  });

  it(`should display help`, () => {
    let result = command.help();
    result = result.replace(/\d+\.\d+\.\d+/, '<version>');
    expect(result).toMatchSnapshot();
  });

  it(`should display all commands`, () => {
    const result = commandHelp();
    expect(result).toMatchSnapshot();
  });

  it(`should display help for the help command`, () => {
    const cli = {
      commands,
      input: ['help', 'help'],
    };
    const stdout = false;
    let result = command.run(cli, { stdout });
    result = result.replace(/\d+\.\d+\.\d+/, '<version>');

    expect(result).toMatchSnapshot();
  });

  it(`should display help for the teach command`, () => {
    const cli = {
      commands,
      input: ['help', 'teach'],
    };
    const stdout = false;
    let result = command.run(cli, { stdout });
    result = result.replace(/\d+\.\d+\.\d+/, '<version>');

    expect(result).toMatchSnapshot();
  });

  it(`should throw for unknown command`, () => {
    const cli = {
      commands,
      input: ['help', 'unknown'],
    };
    const stub = () => command.run(cli);

    expect(stub).toThrow(HelpCommandError);
  });
});
