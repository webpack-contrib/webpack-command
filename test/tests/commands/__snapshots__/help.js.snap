// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help command > should display all commands #0 1`] = `
"  help
  teach"
`;

exports[`help command > should display help #0 1`] = `
"
  [34mhelp[39m v<version>

  Displays help for a given installed webpack command.

  [4mUsage[24m
    $ webpack help <command>

  [4mExamples[24m
    $ webpack help
    $ webpack help init
    $ webpack help serve
"
`;

exports[`help command > should display help #1 1`] = `
"
  [34mhelp[39m v<version>

  Displays help for a given installed webpack command.

  [4mUsage[24m
    $ webpack help <command>

  [4mExamples[24m
    $ webpack help
    $ webpack help init
    $ webpack help serve
"
`;

exports[`help command > should display help for the help command #0 1`] = `
"
  [34mhelp[39m v<version>

  Displays help for a given installed webpack command.

  [4mUsage[24m
    $ webpack help <command>

  [4mExamples[24m
    $ webpack help
    $ webpack help init
    $ webpack help serve
"
`;

exports[`help command > should display help for the teach command #0 1`] = `
"
  [34mteach[39m v<version>

  Teaches webpack-command that a command has been installed and is available.

  [4mUsage[24m
    $ webpack teach --command <command> --module <module>

  [4mOptions[24m
    --command   The name of a command that users will type
    --forget    Instructs the tool to forget a previously added command
    --module    The npm module name of a command

  [4mExamples[24m
    $ webpack teach --command init --module webpack-command-init
    $ webpack teach --command init --forget
"
`;

exports[`lib/flags > should display all commands #0 1`] = `
"  help
  teach"
`;

exports[`lib/flags > should display help #0 1`] = `
"
  [34mhelp[39m v0.0.0

  Displays help for a given installed webpack command.

  [4mUsage[24m
    $ webpack help <command>

  [4mExamples[24m
    $ webpack help init
    $ webpack help serve
"
`;

exports[`lib/flags > should display help for the help command #0 1`] = `
"
  [34mhelp[39m v0.0.0

  Displays help for a given installed webpack command.

  [4mUsage[24m
    $ webpack help <command>

  [4mExamples[24m
    $ webpack help init
    $ webpack help serve
"
`;

exports[`lib/flags > should display help for the teach command #0 1`] = `
"
  [34mteach[39m v0.0.0

  Teaches webpack-command that a command has been installed and is available.

  [4mUsage[24m
    $ webpack teach --command <command> --module <module>

  [4mOptions[24m
    --command   The name of a command that users will type
    --forget    Instructs the tool to forget a previously added command
    --module    The npm module name of a command

  [4mExamples[24m
    $ webpack teach --command init --module webpack-command-init
"
`;
